### ISSUE TITLE
PRINTER MODUE

### ISSUE THREAD

**Issue by MAPA2023** [moneymanagerex/moneymanagerex#6588]
MMEX version:
 - [ ] 1.6.4
 - [] 1.7.0
 - [x ] 1.7.1
 - [ ] 1.7.2
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the OPTIMIZATION
Currently TAGS are not included in prints. It would be interesting to have them.

The high point is the PRINTING FORM, perhaps you can optimize it with a more performing component which for example takes into account the possibility of not printing certain columns.

Please describe your issue with details.
Add screenshot or other files if needed.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

Expected result:

Actual result:

# Additional information

If you have a backtrace for a crash or a warning, paste it here.

**Comment by n-stein** [comment]
What are you printing and how are you attempting to print it?

File -> Print includes tags in transaction panel:

![image](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/0fcdf21a-0f07-476f-b61b-2b03b361e8ab)

and transaction report:

![image](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/54238f14-9ccf-48e2-9364-abebd7ac1010)

**Comment by MAPA2023** [comment]
Hi Stein, 

Thank you for your quick answer. I checked and indeed the TAG column is present. So good! My note would be to find a more flexible "print module" that for example allows you to remove columns that are not necessary for printing. Keep this in mind, maybe there are better performing tools.

**Comment by whalley** [comment]
Yes, at present what is displayed is what is printed. If you omit columns from the transaction display or report then they will not be printed.

**Comment by n-stein** [comment]
> at present what is displayed is what is printed. If you omit columns from the transaction display or report then they will not be printed.

Is this accurate for the checking panels? We don't truly remove a column when it is hidden, we just set its width to 0. So when the table is printed those columns are still included. 

I didn't see a simple way to print a subset of columns, at least on Windows.

**Comment by whalley** [comment]
> Is this accurate for the checking panels? We don't truly remove a column when it is hidden, we just set its width to 0. So when the table is printed those columns are still included.

Good point, I should have  tried it first :-)

**Comment by MAPA2023** [comment]
@whalley @n-stein I did some tests and even though I removed some columns they still printed. In my case the component doesn't work well?

**Comment by n-stein** [comment]
> I did some tests and even though I removed some columns they still printed

Yes, that is what I was describing. Currently hiding columns _in the checking panel_ does not prevent them from printing. However, it does work as expected in the Transaction Report (hidden columns are not printed).

> Possible?

Yes, we have to change from generating columns of width 0 to not generating columns at all when the user hides them.

**Comment by MAPA2023** [comment]
> > I did some tests and even though I removed some columns they still printed
> 
> Yes, that is what I was describing. Currently hiding columns _in the checking panel_ does not prevent them from printing. However, it does work as expected in the Transaction Report (hidden columns are not printed).
> 
> > Possible?
> 
> Yes, we have to change from generating columns of width 0 to not generating columns at all when the user hides them.
@n-stein 
Very good, I'm very pleased, I think it's a very important added value to the product. Prints are important and not necessarily on paper for example on PDF. Being able to modify them and merge only the necessary columns and perhaps "save" the print templates would be very nice.

**Comment by MAPA2023** [comment]
@n-stein @whalley 
Let me know when you implement the change. Good job everyone.

**Comment by MAPA2023** [comment]
@n-stein Hi, have any update on possibility to print only the colum showed ?

**Comment by n-stein** [comment]
No update at this time. This will not be included in 1.7.1.

**Comment by MAPA2023** [comment]
Hi Stein,

fine let me know in which version will be available.

Best.

M.
________________________________
Da: n-stein ***@***.***>
Inviato: martedì 9 aprile 2024 01:08
A: moneymanagerex/moneymanagerex ***@***.***>
Cc: MAPA2023 ***@***.***>; Author ***@***.***>
Oggetto: Re: [moneymanagerex/moneymanagerex] PRINTER MODUE (Issue #6588)


No update at this time. This will not be included in 1.7.1.

**Comment by MAPA2023** [comment]
Hello everybody,

I ask you again when it will be possible to have a printing module with a little more perforation that allows you to remove the columns that are not needed and perhaps also save various pre-set print templates. Thanks as always to everyone.

**Comment by MAPA2023** [comment]
Hi Develope Team, any news on PRINTER MODULE?

**Comment by whalley** [comment]
No, no immediate plans to implement any changes to support this.

**Comment by MAPA2023** [comment]
Hi all new on PRINT MODULE?
